code/online_challenges/project_euler/problem_036/problem_036.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <string>
+
+std::string convertBase2(int number);
+bool isPalindrome(const std::string& str);
+
+int main()
+{
+    int sum = 0;
+    
+    for (int i = 1; i < 1000000; ++i)
+    {
+        if (isPalindrome(std::to_string(i)) && isPalindrome(convertBase2(i)))
+            sum += i;
+    }
+    std::cout << sum << "\n";
+}
+
+std::string convertBase2(int number)
+{
+    std::string result;
+    
+    while (number > 0)
+    {
+        int currentBit = number % 2;
+        int currentQuotient = number / 2;
+        result += std::to_string(currentBit);
+        number = currentQuotient;
+    }
+    
+    return std::string{ result.rbegin(), result.rend() };
+}
+
+bool isPalindrome(const std::string& str)
+{
+    return str == std::string{ str.rbegin(), str.rend() };
```

Comment:
There is a implementation for palindrome check, if the string construct implementation is not lazy.
https://github.com/OpenGenus/cosmos/blob/master/code/utility/palindrome/palindrome_check/palindrome_check.cpp

Reply:
So you are suggesting that I include that file and use that implementation?

Reply:
It's just that I wanted use existed utility of project.
I also tested, both of implementation are same of fast.

Reply:
Ok, then I will #include that file and use it.

Reply:
Ok, or extract header file, it's up to you.